scripts/subsystem_versions.py

```diff
@@ -0,0 +1,69 @@
+"""This script is used to get the versions of the submodules."""
+
+import re
+import sys
+import json
+import argparse
+import subprocess
+
+VERSION_REGEX = re.compile("v([0-9])")
+SUBSYSTEMS = [
+    "head",
+    "gantry-x",
+    "gantry-y",
+    "gripper",
+    "pipettes-single",
+    "pipettes-multi",
+    "pipettes-96",
+    "pippetes-384",
+]
+
+
+def main(args):
+    version_info = {}
+    for subsystem in SUBSYSTEMS:
+        if args.target == "all" or subsystem in args.target:
+            try:
+                version = (
+                    subprocess.check_output(
+                        ["git", "describe", "--tags", "--always", "--match=v*"]
+                    )
+                    .decode()
+                    .strip()
+                )
+                # version should be int
+                version = int(re.search(VERSION_REGEX, version).group(1)) if re.search(VERSION_REGEX, version) else 0
```

Comment:
We could probably pull this into a function that passes in the target we're looking for.

That way, you could use the function to build a version for a single target or for all the subsystems.